From 0e54169d37485a0298535d9ac53055950e1d1382 Mon Sep 17 00:00:00 2001
From: Hector Martin <marcan@marcan.st>
Date: Thu, 3 Mar 2022 02:51:10 +0900
Subject: [PATCH 3/3] autodetect: Look up modules in the target kernel version

Signed-off-by: Hector Martin <marcan@marcan.st>
---
 functions          | 2 +-
 install/autodetect | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/functions b/functions
index e90c7430ea7d..bcf2a42706ca 100644
--- a/functions
+++ b/functions
@@ -304,7 +304,7 @@ auto_modules() {
 
     mapfile -t mods < <(find /sys/devices -name uevent \
         -exec sort -u {} + | awk -F= '$1 == "MODALIAS" && !_[$0]++')
-    mapfile -t mods < <(modprobe -qaR "${mods[@]#MODALIAS=}")
+    mapfile -t mods < <(modprobe -S "$1" -qaR "${mods[@]#MODALIAS=}")
 
     (( ${#mods[*]} )) && printf "%s\n" "${mods[@]//-/_}"
 }
diff --git a/install/autodetect b/install/autodetect
index 846aa3a6d8ae..8297ad00671d 100644
--- a/install/autodetect
+++ b/install/autodetect
@@ -24,7 +24,7 @@ build() {
         return 1
     fi
 
-    mapfile -t mods < <(auto_modules)
+    mapfile -t mods < <(auto_modules "$KERNELVERSION")
     for m in "${mods[@]}"; do
         _autodetect_cache["$m"]=1
     done
-- 
2.35.1

